Allow configure services for individual attachments (#34935)
rails/rails: guides/source/active_storage_overview.md

user.avatar.attached?
```

In some cases you might want to override a default service for a specific attachment.
It can be done in a following way:

```ruby
class User < ApplicationRecord
  has_one_attached :avatar, service: :s3
end
```

Since the provided value would be stored in the DB, there is a possible situation when
configuration for the required service is missing. There is a special configuration option
`Rails.application.configuration.active_storage.service_misconfiguration_behavior` to control
this behavior: `:log` (default) puts the warning to the log, `:fallback_to_default` would try
to use service from `Rails.application.config.active_storage.service`, `:raise` would cause a
`RuntimeError`.

NOTE: if you've initialized ActiveStorage before this feature was shipped, Rails would

> ask you to generate and execute one more migration (`rails active_storage:install && rails db:migrate`).

Reviewer: We can remove this `NOTE` when follow this approach https://github.com/rails/rails/pull/34935/files#r248751180

Author: ✔️

user.avatar.attached?
```

In some cases you might want to override a default service for a specific attachment.

> It can be done in a following way:

Reviewer 2: ```suggestion
You can configure specific services per attachment using the `service` option:
```

```ruby
class User < ApplicationRecord
  has_one_attached :avatar, service: :s3
end
```

Since the provided value would be stored in the DB, there is a possible situation when
configuration for the required service is missing. There is a special configuration option

> `Rails.application.configuration.active_storage.service_misconfiguration_behavior` to control

Reviewer 2: Let's use `Rails.application.config.*` when mentioning configuration.

@message.images.attached?
```

> Overriding the default service can be done in a similar way it was done for `has_one_attached`:

Reviewer 2: ```suggestion
Overriding the default service is done done the same way as `has_one_attached`, by using the `service` option:
```
